src/features/delegation/helpers/retrieve.ts

```diff
@@ -76,27 +71,21 @@ export default function retrieve({
 
             yield put(DelegationAction.setPrivateKey(privateKey))
 
-            dismissToast = false
-
-            tc = yield retoast(tc, {
+            yield toast.open({
                 title: i18n('delegationToast.successLabel'),
                 type: ToastType.Success,
             })
 
             return address
         } catch (e) {
-            dismissToast = false
-
-            tc = yield retoast(tc, {
+            yield toast.open({
                 title: i18n('delegationToast.failureLabel'),
                 type: ToastType.Error,
             })
 
             throw e
         } finally {
-            if (dismissToast) {
-                tc?.dismiss()
-            }
+            yield toast.dismiss({ asap: yield cancelled() })
```

Comment:
If the action was cancelled the toasts will go away ASAP. It does not mean immediately, per se. Inspect the code for `dismiss` to know more.